xbmc/utils/ProgressJob.cpp

```diff
@@ -180,10 +180,20 @@ void CProgressJob::MarkFinished()
   if (m_progress != NULL)
   {
     if (m_updateProgress)
+    {
       m_progress->MarkFinished();
+      //We don't own this pointer and it will be deleted after it's marked finished
+      //just set it to nullptr so we don't try to use it again
+      m_progress = nullptr;
+    }
   }
   else if (m_progressDialog != NULL && m_autoClose)
+  {
     m_progressDialog->Close();
+    //We don't own this pointer and it will be deleted after it's marked finished
+    //just set it to nullptr so we don't try to use it again
+    m_progressDialog = nullptr;
```

Comment:
Can't say for sure now, shouldn't be any dialog calls after markfinished is called if its set to auto close. I do think this is correct but it may need changes elsewhere that i missed. I'm back home monday so I'll go over it again by then.

This shouldn't cause any Isengaard issues unless building from head as this was introduced after 15.0 release.